screens/Discover.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { FC, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { StyleSheet } from 'react-native';
+import { AppState } from '../redux/appState';
+import { ScrollView } from 'react-native-gesture-handler';
+import SwitchSelector from 'react-native-switch-selector';
+import CategoryGrid from '../components/CategoryGrid';
+import StatusBar from '../components/StatusBar';
+import { Text, View } from '../components/Themed';
+import { darkGray, gray, white } from '../theme/colors';
+import { getUserCoordinates, getUserLocation } from '../redux/actions/userDetailsActions';
+import { getRestaurants } from '../redux/actions/restaurantsActions';
+
+const options = [
+  {
+    label: 'DISCOVER',
+    value: 'Discover',
+  },
+  {
+    label: 'NEARBY',
+    value: 'Nearby',
+  },
+];
+
+const Home: FC = () => {
+  const location = useSelector((state: AppState) => state.userDetails?.location);
+  const coordinates = useSelector((state: AppState) => state.userDetails?.coordinates);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(getUserCoordinates());
+  }, []);
+  useEffect(() => {
+    if (coordinates?.latitude && coordinates?.longitude) {
```

Comment:
Remove optional chaining within useSelector hooks and here after initialState and reducers are updated